Objects/frameobject.c

```diff
@@ -747,7 +729,8 @@ static PyMethodDef frame_methods[] = {
 PyTypeObject PyFrame_Type = {
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
     "frame",
-    sizeof(PyFrameObject),
+    offsetof(PyFrameObject, _f_frame_data) +
```

Comment:
This is dangerous: if we ever modify `PyFrameObject` or `InterpreterFrame` these will go out of sync immediately, no?

Reply:
Also, I am confused, how is this accounting for the extra memory needed for the `InterpreterFrame->localsplus` array?

Reply:
The base size of the frameobject is the size of the frameobject without the interpreter frame, plus the base size of the interpreter frame.
Because the  interpreter frame is variable sized, it must be last in the frameobject.
Thus, the base size is `offsetof(PyFrameObject, _f_frame_data) + offsetof(InterpreterFrame, localsplus)`.

`tp_itemsize` specifies the per-item size needed for variable sized objects.

Reply:
:+1: I assume that if someone places stuff after `_f_frame_data` things will be so broken that this would be the smallest of the problems. Makes sense.